### PR TITLE
More appropriate error messages for invalid / unresolvable templates…

### PR DESCRIPTION
**Pull request by seancpeters**
… names

### CODE REVIEW

**Comment by seancpeters** [comment]
Addresses #198

File: src/Microsoft.TemplateEngine.Cli/New3Command.cs [hunk 1]
```diff
@@ -828,6 +836,72 @@ private void ShowTemplateHelp()
             ParameterHelp(allParams, additionalInfo, _hostSpecificTemplateData.HiddenParameterNames);
         }
 
+        private void ShowTemplateNameMismatchHelp()
+        {
+            if (_matchedTemplates.Count == 0)
+            {
+                Reporter.Error.WriteLine("No installed template's names were similiar enough.");
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs]: Grammar seems off, needs to be localizable

File: src/Microsoft.TemplateEngine.Cli/New3Command.cs [hunk 2]
```diff
@@ -828,6 +836,72 @@ private void ShowTemplateHelp()
...
+            {
+                if (template.Info.Tags != null && template.Info.Tags.TryGetValue("type", out string type))
+                {
+                    if (string.Equals(type, "item"))
+                    {
+                        Reporter.Error.WriteLine($"\t- [{template.Info.Name}] is an item template. By default it's only created in a target location containing a project. Force creation with the -all flag.");
+                    }
+                    else
+                    {   // project template
+                        Reporter.Error.WriteLine($"\t- [{template.Info.Name}] is a project template. By default it's not created in a target location containing a project. Force creation with the -all flag.");
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs]: Two dashes in the all switch iirc, loc for these too

Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs]: It's a single dash: "-all" is the short version, the long version is "--show-all"

File: src/Microsoft.TemplateEngine.Cli/New3Command.cs (hunk 1 above)
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs]: This message got removed, as it's effectively a duplicate of the later message indicating there were no matches, i.e.

> Reporter.Error.WriteLine($"No templates matched the input template name: [{_templateName}]");

**Comment by seancpeters** [comment]
The second commit also cleans up the partial-match output, which was originally quite ugly.

File: src/Microsoft.TemplateEngine.Cli/New3Command.cs (hunk 2 above)
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Cli/New3Command.cs]: ah, my bad

**Pull request merged**